pkgs/build-support/go/README.md

```diff
@@ -0,0 +1,42 @@
+## Go toolchain/builder upgrade policy
+
+Go promises that "programs written to the Go 1 specification will continue to compile and run correctly, unchanged, over the lifetime of that specification" [1].
+Newer toolchain versions should build projects developed against older toolchains without problems.
+
+There are however Go packages depending on internal APIs of the toolchain/runtime/stdlib that are not covered by the Go compatibility promise.
+These packages may break on toolchain updates.
+We name these packages `toolchain-breaking`.
+
+There is another set of packages that depends on the toolchain, but in another way:
+Packages providing development support for the Go language (like `gopls`, `golangci-lint`,...) must be compiled with the version they should be used for.
+If `gopls` is compiled for Go 1.23, it won't work for projects that require Go 1.24.
+We name these packages `toolchain-latest`.
```

Comment:
Not quite clear what "these" refers to (1.24, 1.23, ...)

Reply:
I tried to clarify this. "these" aren't the Go toolchains (1.24,1.23,..) in this case, but the packages depending on being built with the latest version (whatever version that currently is).